raiden/api/python.py

```diff
@@ -645,6 +652,30 @@ def transfer_async(
         if not is_binary_address(target):
             raise InvalidAddress('target address is not valid.')
 
+        if secret is not None:
+            if len(secret) != 66:
+                raise InvalidSecretOrSecretHash('secret length should be 66.')
+            if not is_hex(secret):
+                raise InvalidSecretOrSecretHash('provided secret is not an hexadecimal string.')
+            secret = to_bytes(hexstr=secret)
+
+        if secret_hash is not None:
+            if len(secret_hash) != 66:
+                raise InvalidSecretOrSecretHash('secret_hash length should be 64.')
```

Comment:
`random_secret` that exists in Raiden returns 32 bytes. `keccak` from `eth_hash` also returns 32 bytes. Why are you checking for 66 bytes here? Also the error message the length should be 64.

Reply:
The check is for 66 chars not 66 bytes. As explained above - 32 bytes -> 64 string chars. add the '0x' prefix to get to 66,

Error message - will be fixed.